esp-radio/src/ble/controller/mod.rs

```diff
@@ -31,14 +33,16 @@ impl<'d> BleConnector<'d> {
     /// Create and init a new BLE connector.
     #[instability::unstable]
     pub fn new(
-        _init: &'d Controller<'d>,
         device: crate::hal::peripherals::BT<'d>,
         config: Config,
     ) -> Result<BleConnector<'d>, InvalidConfigError> {
         config.validate()?;
+        let _radio_controller = RadioRc::init().map_err(|_init_err| InvalidConfigError)?;
```

Comment:
We should not overwrite the init error with a completely unrelated one. I guess this function should return `enum BleInitError { RadioInit(...), Config(InvalidConfigError) }`.